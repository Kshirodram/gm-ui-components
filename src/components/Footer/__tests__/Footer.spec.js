import React from 'react';
import styled from 'styled-components';
import renderer from 'react-test-renderer';
import Footer, { Wrapper, LongLogo, Copyright, Social } from '../Footer';
import { FONT_GROUP_MAIN_TEXT } from '../../../style/fonts/fontVariables';
import 'jest-styled-components';

let FooterTree, Logo, SocialIcon, CopyrightLink;

beforeAll(() => {
  FooterTree = renderer.create(<Footer />).toJSON();
  Logo = renderer.create(<LongLogo />).toJSON();
  SocialIcon = renderer.create(<Social />).toJSON();
  CopyrightLink = renderer.create(<Copyright />).toJSON();
});

describe('<Footer> Snapshot', () => {
  test('renders correctly', () => {
    expect(FooterTree).toMatchSnapshot();
  });
});

const hover = { modifier: ':hover' };

describe('<Footer> elements', () => {
  test('<Logo> has rule to transition to color on hover', () => {
    expect(Logo).toHaveStyleRule('opacity', '1', hover);
    expect(Logo).toHaveStyleRule('transition', 'all 0.2s ease', hover);
    expect(Logo).toHaveStyleRule('filter', 'grayscale(0%)', hover);
  });

  test('<Social> imgs have rule to transition opacity on hover', () => {
    expect(SocialIcon).toHaveStyleRule('opacity', '1', hover);
  });

  test('<Copyright> is using the correct font-family', () => {
    expect(CopyrightLink).toHaveStyleRule('font-family', FONT_GROUP_MAIN_TEXT);
  });
});
