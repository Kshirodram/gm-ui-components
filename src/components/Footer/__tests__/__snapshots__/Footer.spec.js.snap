// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> Snapshot renders correctly 1`] = `
.c0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  overflow: hidden;
  padding: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: model:rgb;
  color: 68,68,68;
  valpha: 1;
  font-size: 11px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 32px;
  -ms-flex: 0 0 32px;
  flex: 0 0 32px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  text-align: left;
  padding: 0 8px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  opacity: 0.8;
  -webkit-filter: grayscale(100%);
  filter: grayscale(100%);
}

.c1:hover {
  opacity: 1;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-filter: grayscale(0%);
  filter: grayscale(0%);
}

.c1 img {
  max-width: 157px;
}

.c2 {
  margin: 0;
  text-transform: uppercase;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  text-align: center;
  -webkit-transition: opacity 0.2s ease;
  transition: opacity 0.2s ease;
  cursor: default;
  font-family: Nunito;
  z-index: -1;
  opacity: 0.8;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  text-align: right;
  padding: 0 0 0 8px;
}

.c4 {
  padding: 0 8px;
}

.c5 {
  width: 16px;
  opacity: 0.4;
  -webkit-transition: opacity 0.2s ease;
  transition: opacity 0.2s ease;
}

.c5:hover {
  opacity: 1;
}

@media all and (max-width:600px) {
  .c2 {
    display: none;
  }
}

<footer
  className="c0"
>
  <a
    className="c1"
    href="http://deciphernow.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Decipher Technology Studios"
      src=""
    />
  </a>
  <p
    className="c2"
  >
    Â©2017 Decipher Technology Studios
  </p>
  <div
    className="c3"
  >
    <a
      className="c4"
      href="http://github.com/DecipherNow"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="GitHub"
        className="c5"
        src=""
      />
    </a>
    <a
      className="c4"
      href="http://twitter.com/deciphernow"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Twitter"
        className="c5"
        src=""
      />
    </a>
    <a
      className="c4"
      href="http://www.linkedin.com/company/decipher-technology-studios"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="LinkedIn"
        className="c5"
        src=""
      />
    </a>
  </div>
</footer>
`;
